internal/bgp/bgpfrr/frr.go

```diff
@@ -0,0 +1,150 @@
+package bgpfrr
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"go.universe.tf/metallb/internal/bgp"
+)
+
+// As the MetalLB controller should handle messages synchronously, there should
+// no need to lock this data structure. TODO: confirm this.
+type frrState struct {
+	sessions map[string]*Session
+	config   *frrConfig
+}
+
+var state *frrState
+
+func init() {
+	state = &frrState{
+		sessions: map[string]*Session{},
+	}
+}
+
+type Session struct {
+	asn        uint32
```

Comment:
I have a general comment about this elsewhere so I will close this as a duplicate. I want to go through all the MetalLB speaker code because I *think* I saw different conventions in different places. I need to confirm. However, I think the `MyASN` `ASN` convention is the best.